jet/cdc/src/main/java/com/hazelcast/samples/jet/cdc/CdcRealTimeAnalysisWithParallelSnapshotDemo.java

```diff
@@ -0,0 +1,173 @@
+        var config = new Config();
+        config.getJetConfig().setEnabled(true);
+        config.getSerializationConfig()
+                .getCompactSerializationConfig()
+                .register(CustomerStatsReport.class)
+                .setEnabled(true);
+
+        HazelcastInstance hz = Hazelcast.newHazelcastInstance(config);
+        JetService jet = hz.getJet();
+        DriverManager.getDrivers();
+        // trigger initialization
+        hz.getMap("CustomerStatsReport");
+
+        hz.getSql().execute(
+          """
+                  create mapping CustomerStatsReport (
+                    customerId int,
+                    customerFirstName varchar,
+                    customerLastName varchar,
+                    ordersTotal int,
+                    itemsTotal int,
+                    itemsAvg double)
+                  type IMap options (
+                    'keyFormat' = 'int',
+                    'valueFormat' = 'compact',
+                    'valueCompactTypeName'='CustomerStatsReport'
+                  );
+                  """
+        );
+
+        Pipeline pipeline = Pipeline.create();
+        var orderJdbc = pipeline.readFrom(
+                jdbc(
+                        () -> DriverManager.getConnection("jdbc:postgresql://localhost:5432/postgres", "postgres", "cdcpwd1337"),
+                        (connection, parallelism, index) -> {
+                            String sql = "select * from inventory.orders where id % ? = ?";
+                            PreparedStatement stmt = connection.prepareStatement(sql);
+                            stmt.setInt(1, parallelism);
+                            stmt.setInt(2, index);
+                            return stmt.executeQuery();
+                        },
+                        rs -> {
+                            Order order = new Order(
+                                    rs.getInt("id"),
+                                    rs.getDate("order_date"),
+                                    rs.getInt("purchaser"),
+                                    rs.getInt("product_id"),
+                                    rs.getInt("quantity")
+                            );
+                            return orderEvent(order, Operation.SYNC);
+                        })).addTimestamps(o -> o.event.orderDate().getTime(), LAG);
+
+        var cdc = pipeline.readFrom(
+                        postgres("inventory-db")
+                                .setDatabaseName("postgres")
+                                .setSchemaWhitelist("inventory")
+                                .setDatabaseAddress("localhost")
+                                .setDatabaseUser("postgres")
+                                .setDatabasePassword("cdcpwd1337")
+                                .setTableWhitelist("inventory.orders", "inventory.customers")
+                                .setDatabasePort(5432)
+                                .setCustomProperty("snapshot.mode", "initial")
```

Comment:
If I understand correctly this will run initial snapshot, which you also do in parallel with the jdbc source. Did you mean to use other value here?

Reply:
This is a tricky one. We are streaming the Debezium's snapshot data, but also reading them ourselves. The "slower" data will be discarded later in the mapStateful. This will make reading faster, but still not optimal (for optimal reading we would need to modify Debezium).

It would be better to have separate values for each table, but debezium doesn't support that and we can create only 1 source for one database (otherwise Debezium fails).